src/quo2/components/messages_gap.cljs

```diff
@@ -0,0 +1,155 @@
+
+(defn on-press
+  [chat-id gap-ids]
+  (fn []
+    (>evt [:chat.ui/fill-gaps chat-id gap-ids])))
+
+;;; components
+;;;; borders
+(defn hborder [{:keys [type style]}]
+  [rn/image {:source      (get-image :horizontal)
+             :resize-mode :repeat
+             :style       (merge {:position           :absolute
+                                  :left               0
+                                  :padding-horizontal 4
+                                  :overflow           :hidden
+                                  :width              "110%"
+                                  :height             8
+                                  :margin-left        -4}
+                                 (if (= type :top)
+                                   {:top 0}
+                                   {:transform [{:rotateZ "180deg"}]
+                                    :bottom    0})
+                                 style)}])
+
+(defn vborder [type body-height]
+  (let [height @body-height
+        img    (get-image :vertical)]
+    (when (and img height)
+      [rn/image {:source      img
+                 :resize-mode :repeat
+                 :style       (merge
+                               {:position :absolute
+                                :top      4
+                                :height   (- height 8)
+                                :width    4}
+                               (if (= type :left)
+                                 {:left 0}
+                                 {:transform [{:rotate "180deg"}]
+                                  :right     0}))}])))
+
+;;;; others
+(defn circle []
+  [rn/view
+   {:width         8
+    :height        8
+    :border-width  1
+    :margin        4
+    :flex          0
+    :border-color  (get-color :icon)
+    :border-radius 50}])
+
+(defn timestamp [str]
+  [text/text {:size  :label
+              :style {:text-transform :none
+                      :color          (get-color :time)}} str])
+
+(defn info-button [on-press]
+  [pressable-class
+   {:on-press on-press}
+   [rn/image {:source info-img :width 12 :padding 4}]])
+
+;;;; left/right
+(defn left []
+  [rn/view {:flex            0
+            :padding-left    11.5
+            :margin-right    20.5
+            :align-items     :center
+            :justify-content :space-between}
+   [circle]
+   [rn/image {:style {:flex 1} :source (get-image :circles) :resize-mode :repeat}]
+   [circle]])
+
+(defn right [timestamp-far timestamp-near chat-id gap-ids on-info-button-pressed]
+  [rn/view {:flex 2}
+   [rn/view
+    {:flex-direction  :row
+     :align-items     :center
+     :justify-content :space-between
+     :margin-right    2}
+    [timestamp timestamp-far]
+    (and (fn? on-info-button-pressed)
+         [info-button on-info-button-pressed])]
+
+   [pressable-class
+    {:style    {:flex 2 :margin-top 16}
+     :on-press (when (and chat-id gap-ids)
+                 (on-press chat-id gap-ids))}
```

Comment:
doesn't look like it worth it to have on-press it could be just 
```clojure 
#(when (and chat-id gap-ids)
                 (>evt [:chat.ui/fill-gaps chat-id gap-ids]))
```

Reply:
it much simpler to read event in the view , better to do not hide them